dev/NavigationView/NavigationView.cpp

```diff
@@ -2264,9 +2279,20 @@ void NavigationView::UpdateSingleSelectionFollowsFocusTemplateSetting()
 
 void NavigationView::OnSelectedItemPropertyChanged(winrt::DependencyPropertyChangedEventArgs const& args)
 {
+
     auto newItem = args.NewValue();
+    auto oldItem = args.OldValue();
 
-    ChangeSelection(args.OldValue(), newItem);
+    ChangeSelection(oldItem, newItem);
+
+    // Animate to be sure the selected item is visually higlighted!
+    // See #1395 for additional context
+    if (oldItem != newItem)
+    {
+        ChangeSelectStatusForItem(oldItem, false /*selected*/);
+        ChangeSelectStatusForItem(newItem, true /*selected*/);
+        AnimateSelectionChanged(oldItem, newItem);
+    }
```

Comment:
I'm still not sure I undertand why this is needed, and looking through the comment history on that pull request I'm not sure the discussion ever came to a conclusion.  maybe @Ojhad has more context and can review.

Reply:
The problem was that, when we prevent SelectionChanged from being raised, the new selected item does not get animated and the visual indicator which item is selected is missing. Thus we need to make sure we animate it.

Reply:
I think we are still at the same problem as mentioned in the previous discussion. 
Having this code block here seems like a less than ideal solution because we are executing this for every selection change, which unnecessarily duplicates calls for the majority of scenarios, but also, as mentioned in the previous discussion, this may cause issues due to some necessary code in `ChangeSelection` not being executed.

Another issue, Top Nav and Left Nav have different settings items. So by not raising the SelectionChanged event at all, the user can get into a state where they are holding on to the wrong settings item. Due to this, it seems like the best solution would be to ensure that the selected property is never passed as null rather than blocking all the SelectionChanged events.

Reply:
> I think we are still at the same problem as mentioned in the previous discussion.
Having this code block here seems like a less than ideal solution because we are executing this for every selection change, which unnecessarily duplicates calls for the majority of scenarios, but also, as mentioned in the previous discussion, this may cause issues due to some necessary code in ChangeSelection not being executed.

Since we need this code only if we skip `ChangeSelection` which will only be skipped when we set `m_shouldIgnoreNextSelectionChange`, we could simply check , and only run the code if it is set.

> Another issue, Top Nav and Left Nav have different settings items. So by not raising the SelectionChanged event at all, the user can get into a state where they are holding on to the wrong settings item. Due to this, it seems like the best solution would be to ensure that the selected property is never passed as null rather than blocking all the SelectionChanged events.

Yes this is really a problem. However, in my opinion, raising SelectionChanged because the displaymode changed is not the behavior one would expect. In what way is the user able to hold onto the settings item? Are we actually exposing the UI item or just a reference to a settings item (which I think we change behind the scenes when switching displaymode)?

Reply:
> Yes this is really a problem. However, in my opinion, raising SelectionChanged because the displaymode changed is not the behavior one would expect. 

Yeah, I agree with you that this behavior seems a bit awkward.

> In what way is the user able to hold onto the settings item? Are we actually exposing the UI item or just a reference to a settings item (which I think we change behind the scenes when switching displaymode)?

We pass the selected container in the args object. 
I think the question becomes, are we willing to get into a state where the last container passed in the args object is not the same one being returned by the `SelectedItem` (and `SettingsItem`) API after display mode changes. That seems worse to me. 
In my opinion, the best option for now is to continue raising a SelectionChanged event to ensure that the user gets informed of the correct selected container. If we just prevent the event args from passing a null SelectedContainer, this would fix the scenario described in the linked bug.
Maybe @YuliKl has some thoughts?

Reply:
I got a little lost in the discussion on this PR so went back to the original issue #148 that this is trying to fix. I feel like the important part is that we fix this:
> What's actually happening is that the app is crashing on trying to access the content of the selected item, which is null.

Reply:
Yes the main idea was to fix #148, however since there is now #1956 that, from the posted information, seems to be caused by the behavior of raising `SelectionChange` when switching displaymode, we maybe this is the better choice? 

Otherwise developers need to filter themselves if the app should navigate or not based on previous selcted item.

Reply:
After some discussion, I think it makes most sense to just keep this implementation and block the selection changed events from being raised. If the user needs the updated settings item, they can get it through the NavigationView API. 

@chingucoding - can you update the change so that this fix only gets executed for the settings scenario (as we discussed above)?

Reply:
Sure, I have updated a check to ensure we only call that when the settings item is selected.